jwst/guider_cds/guider_cds.py

```diff
@@ -0,0 +1,133 @@
+#! /usr/bin/env python
+# 
+#  guider_cds.py
+
+from __future__ import division
+import time
+import numpy as np
+import logging
+
+from .. import datamodels
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+def guider_cds(model):
+    """
+    Extended Summary
+    ----------------
+    Calculate the count rate for each pixel in all data cube sections and all
+    integrations.
+
+    For each integration in a given FGS input dataset whose mode is ACQ1, 
+    ACQ2, or TRACK, the count rate is the last group minus the first group, 
+    divided by the effective integration time.  If the mode is ID, the last 
+    group minus the first group is calculated for both integrations; the count 
+    rate is then given by the minimum of these two values for each pixel, 
+    divided by the effective integration time.  For the FINEGUIDE mode, the 
+    count rate is the average of the last 4 groups minus the average of the 
+    first 4 groups, divided by the effective integration time.  
+
+    Parameters
+    ----------
+    model: data model
+        input data model, assumed to be of type FGSModel
+    """
+    # get needed sizes and shapes
+    imshape, n_int, effinttm, readpatt = get_dataset_info(model)
+
+    if readpatt == 'ID': # force output to have single slice
```

Comment:
This could now use the EXP_TYPE keyword. I'm not sure how consistent or useful the readpatt values are guider products.